testkit/src/main/java/org/apache/calcite/sql/test/SqlTestFactory.java

```diff
@@ -130,6 +130,12 @@ protected SqlTestFactory(CatalogReaderFactory catalogReaderFactory,
 
   /** Creates a parser. */
   public SqlParser createParser(String sql) {
+    if (sql.isEmpty()) {
+      // I could not figure out how to convince the grammar to accept an empty
```

Comment:
I'm not very familiar with this place, maybe it's a JavaCC issue as you said, but it does feel strange.